lib/no_brainer/migrator.rb

```diff
@@ -0,0 +1,125 @@
+# frozen_string_literal: true
+
+module NoBrainer
+  #
+  # The Migrator class is used in order to execute a migration script and store
+  # the migration script "version" or timestap in a table allowing to run it
+  # only once.
+  # The mechanism is heavily inspired by the ActiveRecord one.
+  #
+  # It supports both migrate and rollback.
+  #
+  class Migrator
+    def self.migrations_table_name
+      'nobrainer_migration_versions'
```

Comment:
Create a document class for the migration table, similar to lib/no_brainer/lock.rb
You can call it MigrationState, and the table can be called 'nobrainer_migrations'
This should simplify your table manipulation logic.